### ISSUE TITLE
Porting the code to cu113 and learning doesn't seem to stabilize

### ISSUE THREAD

**Issue by xodus777** [MMRobotics/RSLO#5]
Hello,
Thank you for sharing this amazing work! I've been adapting your code to work with the KITTI dataset.
Since the public NVIDIA Docker images no longer provide CUDA 9.2, I've updated the code for compatibility with newer environments. I've tried to keep all critical components unchanged, but I'm experiencing training issues even after 20k iterations. I've also tried to lower learning rates as you've mentioned in this [issue](https://github.com/DecaYale/RSLO/issues/4#issue-2256474282).
However, the training seems to diverge, or at least abnormal, particularly with the rotation component (as shown in the attached graph). Would it be possible to get some guidance on the expected training behavior? For example:

- Training logs or loss curves from successful runs
- Any known sensitive hyperparameters or common issues
- Approximately when the training starts to stabilize

Any insights would be greatly appreciated. Thank you!

<img width="551" height="451" alt="Image" src="https://github.com/user-attachments/assets/a19ee020-c829-4acb-8798-0f31628c8ced" />

<img width="640" height="480" alt="Image" src="https://github.com/user-attachments/assets/5e3ad6f3-2e0e-417b-8fbd-ea790aa1a85a" />

**Comment by DecaYale** [comment]
Thank you for your interest in our work. How large is the batch size, and how many GPUs are you using?  Since this is a project we completed several years ago, I may have forgotten some details. However, I recall that we at least need 8 GPUs for training 200,000 iterations.  The trajectories should look reasonable after 100K iterations.

**Comment by DecaYale** [comment]
Moreover, could you create a pull request to share your updated environments? It could be helpful for the researchers who are interested in this work.

**Comment by xodus777** [comment]
I'm training on an RTX 4090 and have set the batch size to 1, as it looks like part of the network asserts this value ([link to code](https://github.com/DecaYale/RSLO/blob/9ef3689bcc2baa71aea5106e79dd5e6a047ce707/rslo/models/middle.py#L221)).

Once I can confirm everything is working correctly after the fix, I'll create a pull request.

**Comment by DecaYale** [comment]
Then you may need more GPU resources for training--at least 8. Let me know if it still doesn't work
